source/app/documentmanager.cpp

```diff
@@ -120,6 +120,22 @@ int DocumentManager::getCurrentDocumentIndex() const
     return *myCurrentIndex;
 }
 
+size_t DocumentManager::getDocumentListSize() const 
+{
+	return myDocumentList.size();
+}
+
+int DocumentManager::containsDocument(const std::string& filepath)
+{
+	for (int i = 0; i < getDocumentListSize(); ++i)
+	{
+		Document& doc = getDocument(i);
+		if (filepath == doc.getFilename())
```

Comment:
You need to check `doc.hasFilename()` as well, for documents that haven't been saved yet. To test, create a new document and then try to open an existing document.